plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExBudgetingDefaults.hs

```diff
@@ -77,7 +78,89 @@ defaultCekParameters :: MachineParameters CekMachineCosts CekValue DefaultUni De
 defaultCekParameters = toMachineParameters defaultCekCostModel
 
 unitCekParameters :: MachineParameters CekMachineCosts CekValue DefaultUni DefaultFun
-unitCekParameters = toMachineParameters (CostModel unitCekMachineCosts defaultBuiltinCostModel)
+unitCekParameters = toMachineParameters (CostModel unitCekMachineCosts unitCostBuiltinCostModel)
 
 defaultBuiltinsRuntime :: HasConstantIn DefaultUni term => BuiltinsRuntime DefaultFun term
 defaultBuiltinsRuntime = toBuiltinsRuntime defaultBuiltinCostModel
+
+
+-- A cost model with unit costs, so we can count how often each builtin is called
+
+unitCostOneArgument :: CostingFun ModelOneArgument
+unitCostOneArgument =  CostingFun (ModelOneArgumentConstantCost 1) (ModelOneArgumentConstantCost 0)
+
+unitCostTwoArguments :: CostingFun ModelTwoArguments
+unitCostTwoArguments   =  CostingFun (ModelTwoArgumentsConstantCost 1) (ModelTwoArgumentsConstantCost 0)
+
+unitCostThreeArguments :: CostingFun ModelThreeArguments
+unitCostThreeArguments =  CostingFun (ModelThreeArgumentsConstantCost 1) (ModelThreeArgumentsConstantCost 0)
+
+unitCostSixArguments :: CostingFun ModelSixArguments
+unitCostSixArguments   =  CostingFun (ModelSixArgumentsConstantCost 1) (ModelSixArgumentsConstantCost 0)
+
+unitCostBuiltinCostModel :: BuiltinCostModel
+unitCostBuiltinCostModel = BuiltinCostModelBase
+    {
+     -- Integers
+      paramAddInteger               = unitCostTwoArguments
+    , paramSubtractInteger          = unitCostTwoArguments
```

Comment:
Unfortunately I think we just have to repeat the entire thing here.  I initially thought we'd be able to use some map-like thing to modify the default model, but I think that's not going to work because the types of the fields vary.

Reply:
Irritating, but I think not worth worrying about right now indeed.